### PR TITLE
feat(2199): Add endpoints to get and update user settings [3]

### PR DESCRIPTION
**Pull request by tkyi**
## Context

It would be nice if we stored user settings and pipeline settings in the backend.

## Objective

This PR adds endpoints to get and update user settings.
Also allows user to update pipeline settings.

## References

Related to #2199 

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/users/settings/get.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: async (request, h) => {
+            const { userFactory } = request.server.app;
+
+            return userFactory
```
Comment (pritamstyz4ever) [inline plugins/users/settings/get.js:28]: should we use async-await here?

File: test/lib/registerPlugins.test.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ const sinon = require('sinon');
 
 sinon.assert.expose(Assert, { prefix: '' });
 
-describe('Register Unit Test Case', () => {
+describe.only('Register Unit Test Case', () => {
```
Comment (pritamstyz4ever) [inline test/lib/registerPlugins.test.js:9]: remove .only?

File: plugins/users/settings/update.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                        throw boom.forbidden(`User ${username} cannot update user settings for user ${user.username}`);
+                    }
+
+                    return user.updateSettings(settings).then(results => {
+                        return h.response(results).code(200);
+                    });
+                })
+                .catch(err => {
+                    throw err;
+                });
```
Comment (pritamstyz4ever) [inline plugins/users/settings/update.js:47]: change to async await, and remove catch err

Comment (tkyi) [reply plugins/users/settings/update.js]: updated

File: plugins/users/settings/get.js (hunk 1 above)
Comment (tkyi) [reply plugins/users/settings/get.js]: updated

File: plugins/users/settings/update.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            const { userFactory } = request.server.app;
+            const { username } = request.auth.credentials;
+            const { settings } = request.payload;
+            const user = await userFactory.get(request.params.id);
+
+            if (!user) {
+                throw boom.notFound('User does not exist');
+            }
+
+            if (user.username !== username) {
```
Comment (adong) [inline plugins/users/settings/update.js:35]: is username `uuid`?  Curious if two users can have same username?

**Review by adong — APPROVED** [review]

**Pull request merged**